core-library/src/main/java/org/silverpeas/core/contribution/template/publication/PublicationTemplateManager.java

```diff
@@ -510,4 +527,136 @@ protected void registerForRenewingContentCipher() {
     ContentEncryptionServiceProvider.getContentEncryptionService()
         .registerForRenewingContentCipher(contentIterator);
   }
+
+  public void saveData(String xmlFormName, PagesContext context, List<FileItem> items)
+      throws SilverpeasException {
+    if (context == null) {
+      throw new SilverpeasException("Context must be defined !");
+    }
+    if (StringUtil.isNotDefined(context.getComponentId()) ||
+        StringUtil.isNotDefined(context.getObjectId()) ||
+        StringUtil.isNotDefined(context.getLanguage()) ||
+        StringUtil.isNotDefined(context.getUserId())) {
+      throw new SilverpeasException(
+          "Context is not complete ! ComponentId, ObjectId, Language and UserId must be " +
+              "defined...");
+    }
+    if (StringUtil.isDefined(xmlFormName)) {
+      PublicationTemplate pub = getTemplateAndRegisterIt(xmlFormName, context);
+      try {
+        RecordSet set = pub.getRecordSet();
+        Form form = pub.getUpdateForm();
+        DataRecord data = set.getRecord(context.getObjectId());
+        if (data == null) {
+          data = set.getEmptyRecord();
+          data.setId(context.getObjectId());
+        }
+
+        // sauvegarde des données du formulaire
+        form.update(items, data, context);
+        set.save(data);
+      } catch (Exception e) {
+        throw new SilverpeasException("Unable to save data", e);
+      }
+    }
+  }
+
+  private PublicationTemplate getTemplateAndRegisterIt(String xmlFormName, PagesContext context)
+      throws SilverpeasException {
+    String shortName = getShortName(xmlFormName);
+    String externalId = context.getComponentId() + ":" + shortName;
+    PublicationTemplate pub;
+    try {
+      // récupération des données du formulaire (via le DataRecord)
+      pub = getPublicationTemplate(externalId);
+    } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)